src/ch01_arrays_and_strings/iq01.rs

```diff
@@ -33,21 +33,73 @@ pub fn is_unique_no_ds(input: &str) -> bool {
     true
 }
 
-pub const INPUT_EXPECTED: &[(&str, bool)] = &[
-    ("abcde", true),
-    ("hello", false),
-    ("", true),
-    ("Ff", true),
-    ("noon", false),
-    ("non", false),
-    ("Non", true),
-];
+use lazy_static::lazy_static;
+
+fn gen_input_expected() -> &'static [(&'static str, bool)] {
+    let input_expected = &[
+        ("abcde", true),
+        ("hello", false),
+        ("", true),
+        ("Ff", true),
+        ("noon", false),
+        ("non", false),
+        ("Non", true),
+    ];
+
+    input_expected
+}
+
+//lazy_static! {
+//    static ref INPUT_EXPECTED: &'static [(&'static str, bool)] = {
+//        gen_input_expected()
+//    };
+//}
+
+fn gen_global_data() -> &'static [(&'static str, bool)] {
+    let input_expected = &[
+        ("item1", true),
+        ("item2", false),
+        ("item3", true),
+        ("item4", false),
+        ("item5", true),
+    ];
+
+    input_expected
+}
+lazy_static! {
+    static ref GLOBAL_DATA: &'static [(&'static str, bool)] = { gen_input_expected() };
+}
 
 #[cfg(test)]
 mod tests {
     use super::*;
     use gen_tests::generate_tests;
 
+    static INPUT_EXPECTED: &[(&str, bool)] = &[
+        ("abcde", true),
+        ("hello", false),
+        ("", true),
+        ("Ff", true),
+        ("noon", false),
+        ("non", false),
+        ("Non", true),
+    ];
+
+    //const INPUT_EXPECTED: &[(&str, bool)] = &[
+    //    ("abcde", true),
+    //    ("hello", false),
+    //    ("", true),
+    //    ("Ff", true),
+    //    ("noon", false),
+    //    ("non", false),
+    //    ("Non", true),
+    //];
+
+    /*
+    static INPUT_LEN: usize = INPUT_EXPECTED.len();
+    const INPUT_LEN: usize = INPUT_EXPECTED.len();
+    */
+
```

Comment:
TODO: remove comments
It doesn't matter between static or const here. I prefer const, but there should be no leftover code comments